webapp/src/pages/createForm/components/ApplicationForm.js

```diff
@@ -0,0 +1,181 @@
+import React, { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next'
+import { eventService } from '../../../services/events';
+import icon from '../icon.svg';
+import Section from './Section';
+
+const ApplicationForm = (props) => {
+  const [nominate, setNominate] = useState(false);
+  const [event, setEvent] = useState({
+    loading: true,
+    event: null,
+    error: null,
+  });
+  const [sections, setSections] = useState([{
+    id: 1,
+    name: 'Untitled Section',
+    description: '',
+    questions: [
+      {
+        id: `${Math.random()}`,
+        order: 1,
+        headline: '',
+        placeholder: '',
+        type: null,
+        options: [],
+        value: '',
+        label: '',
+        required: false
+      }
+    ]
+  }]);
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    eventService.getEvent(props.event.id).then( res => {
+      setEvent({
+        loading: res.loading,
+        event: res.event,
+        error: res.error
+      })
+    })
+  }, []);
+  const handleCheckChanged = (e) => {
+    setNominate(e.target.checked);
+  }
+  const handleSection = (input) => {
+    setSections(input)
+  }
+  const addSection = () => {
+    const id = sections.length + 1;
+    setSections([...sections, {
+      id: id,
+      name: 'Untitled Section',
+      description: '',
+      questions: [
+        {
+          id: 1,
+          order: 1,
+          headline: '',
+          placeholder: '',
+          type: null,
+          options: [],
+          value: '',
+          label: '',
+          required: false
+        }
+      ]
+    }]);
+  }
+  const addQuestion = (id) => {
+    let newSections = sections.map(e => {
+      if(e.id === id) {
+        return {...e, questions: [...e.questions, {
+          id: `${Math.random()}`,
+          order: e.questions.length && e.questions.length + 1,
+          headline: '',
+          placeholder: '',
+          type: null,
+          options: [],
+          value: '',
+          label: '',
+          required: false
+        }]}
+      }
+      return e;
+    });
+    setSections(newSections);
+  } 
+  const dateFormat = (date) => {
+    return new Date(date).toLocaleDateString('en-US', {
+      weekday: 'short',
+      month: 'long',
+      day: '2-digit',
+      year: 'numeric',
+      hour: '2-digit'
+    })
+  }
+
+  const TopBar = () => {
+    return (
+      <div className="top-bar">
+        <div className="icon-title">
+          <img src={icon} alt="form" className="icon" />
+          <span className="title">{t("Form title")}</span>
+        </div>
+        <button className="create-form-btn">{t("Save")}</button>
+      </div>
+    );
+  }
+  const FormSection = () => {
+    const {loading, event: evnt, error } = event;
+ 
+    if (loading) {
+      return (
+        <div>
+          <i className="fa fa-spinner fa-lg" aria-hidden="true"></i>
```

Comment:
FYI, we have a [loading component](https://github.com/deep-learning-indaba/Baobab/blob/develop/webapp/src/components/Loading.js) which you can use here instead